### PR TITLE
Update developer guide

### LINKED ISSUE #164: Add manual testing instructions to developer guide

### PR DESCRIPTION
**Pull request by teojunda**
**Changes made**:
+ Add manual testing section. Closes #164 
+ Add planned enhancements
+ Update use cases
+ Update personal page

Also, fixed bug where side panel would not be cleared after `clear` command.

### CODE REVIEW

**Review by g-tejas — APPROVED** [review]
CleanShot 2024-04-08 at 15 15 39@2x

LGTM! Checked the website and it's indeed okay to fix this because after a `clear` command, the state is reset and the side panel's showing the data of a deleted contact

File: docs/team/teojunda.md [hunk 1]
```diff
@@ -5,42 +5,40 @@ title: "Axel's Project Portfolio Page"
 
 ### Project: TaToolkit Level 3
 
-TaToolkit - Level 3 is a desktop application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
+TaToolkit is a desktop application used for teaching assistant to manage their students' personal details and other information related to their classroom. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 28 kLoC.
```
Comment (kailashgautham) [inline docs/team/teojunda.md:8]: small typo - teaching assistants *

Comment (kailashgautham) [reply docs/team/teojunda.md:8]: i see my ppp has the same issue, could I trouble you to help me change it :') thankss

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -701,44 +652,153 @@ testers are expected to do more *exploratory* testing.
...
+
+Test case: `exit`
+
+Expected Output: TA Toolkit closes.
 
 --------------------------------------------------------------------------------------------------------------------
 
-## Planned Enhancements
+## Appendix: Planned Enhancements
+
```
Comment (kailashgautham) [inline docs/DeveloperGuide.md:787]: I believe the first line below planned enhancements needs to have team size, as we are only allowed to have TEAM_SIZE * 2 number of planned enhancements i.e. 10 in our case - so if possible would be good to number each enhancement as well, thanks!

smth like 

`Team Size: 5`

**Review by kailashgautham — CHANGES_REQUESTED** [review]
LGTM! Just some minor changes

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (teojunda) [reply docs/DeveloperGuide.md:787]: Oh i see. I was wondering why you put that

**Review by kailashgautham — APPROVED** [review]
LGTM!

**Pull request merged**